vtex/pickup-points-modal
# CHK-904: Fix grouping of opening times not considering fridays

#### What is the purpose of this pull request?

In this PR I made basically two changes, one updating Jest and ESLint 
dependencies, and another was the actual bugfix. I **highly recommend** to 
review these commits individually, else it will be really hard reviewing the 
differences (or you can use GitHub builtin VSCode to review, which would also 
be easier :P).

I haven't fixed all ESLint issues on purpose, as it would make the review a lot 
harder because of prettier.

I also took the opportunity to refact the `PickupPointDetails` and extract the 
piece of UI containing the table for the business hours to a separate 
component, and also rewrote it to TypeScript. The fix is basically just one 
line of code (an off-by-one error) which I'll point out in a comment.

#### What problem is this solving?

In this PR I fixed the issue where the opening and closing times of a pickup 
point were not considering the difference on Fridays. If you had the same 
opening and closing times from Monday through Thursday, the Friday time would 
not be considered and it would show that the times from Monday through Friday 
were the same.

#### How should this be manually tested?

You can check in [this workspace](https://lucas--shoppingsquad.myvtex.com/checkout/cart/add/?sku=2&qty=1&seller=1&sc=1)

Use the postal code `22250-040` and check the pickup point details. You should 
see that the times from Monday through Thursday are the same, but on Friday it 
isn't, and it is properly displayed in the UI.

#### Screenshots or example usage

Before

![image](https://user-images.githubusercontent.com/10223856/140933244-acf34bbb-ac8d-4605-8c34-596c88b02a73.png)

After

![image](https://user-images.githubusercontent.com/10223856/140933181-217dce4d-7e56-491c-a803-64119abc8f4e.png)

#### Types of changes
